client/app/lib/util/verifyPassword.coffee

```diff
@@ -0,0 +1,13 @@
+whoami = require 'app/util/whoami'
+remote = require 'app/remote'
+
+module.exports = (password, callback) ->
+
+  whoami().fetchEmail (err, email) ->
+    options = { password, email }
+    remote.api.JUser.verifyPassword options, (err, confirmed) ->
```

Comment:
we shouldn't require two explicit calls in here I believe, let's update or provide another method for verification password for the active session owner.